File: algos_assignment/3rd_grade/이서진_metamong/2nd_week1/11725.py
import sys
from collections import deque
input=sys.stdin.readline
n=int(input())
tree=[[]for i in range(n+1)]
chk=[False for i in range(n+1)]
parent=[0 for i in range(n+1)]
for i in range(n-1):
    a,b=map(int,input().split())
    tree[a].append(b)
    tree[b].append(a)

q=deque([1])
chk[1]=True

while q:
    now=q.popleft()
    for next in tree[now]:
        if not chk[next]:
            parent[next]=now;
            chk[next]=True
            q.append(next)
for i in range(2,n+1):
    print(parent[i])

File: algos_assignment/2nd_grade/장아현_becherryu/week5/11725.cpp
//dfs 이용 자식노드까지 탐색하기
#include <iostream>
#include <vector>
using namespace std;

int tree[100001];
bool parent[100001];
vector <int> v[100001];

int n;

void dfs(int x)
{	int next;

	parent[x] = true;
	
	for (int i = 0; i < v[x].size(); i++) {
		next = v[x][i];

		if (!parent[next]) {
			tree[next] = x;
			dfs(next);
		}
	}
}

int main()
{
	cin >> n;

	for (int i = 0; i < n; i++) {
		int a, b;
		cin >> a >> b;
		v[a].push_back(b);
		v[b].push_back(a);
	}

	dfs(1);

	for (int i = 2; i <= n; i++) {
		cout << tree[i] << "\n";
	}
}
